From 49d824ef1fdeca5ea5bb28ea6e68a24d21a96756 Mon Sep 17 00:00:00 2001
From: Steve Grubb <ausearch.1@gmail.com>
Date: Wed, 28 Jan 2026 21:29:17 -0500
Subject: [PATCH] Potential memory leak on early-return in file_append
Content-type: text/plain

If add_list_load_path(path) returns failure after adding items to add_list, the function returns without calling list_empty, which can leak any accumulated entries. Call list_empty before returning an error.
---
 src/cli/file-cli.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/cli/file-cli.c b/src/cli/file-cli.c
index 7e02e939fdb0..b07d42ee070c 100644
--- a/src/cli/file-cli.c
+++ b/src/cli/file-cli.c
@@ -130,8 +130,10 @@ int file_append(const char *path, const char *fname, bool use_filter)
 
 	list_init(&add_list);
 	rc = add_list_load_path(path);
-	if (rc)
+	if (rc) {
+		list_empty(&add_list); // could be partially populated by nftw
 		return rc;
+	}
 
 	if (use_filter && filter_prune_list(&add_list, NULL)) {
 		list_empty(&add_list);
-- 
2.53.0

